src/core/transaction.c

```diff
@@ -628,7 +628,9 @@ static int transaction_apply(
                 HASHMAP_FOREACH(j, m->jobs) {
                         assert(j->installed);
 
-                        if (j->unit->ignore_on_isolate)
+                        if (j->unit->ignore_on_isolate ||
+                                        (manager_state(j->unit->manager) != MANAGER_RUNNING &&
+                                        unit_survives_soft_reboot(j->unit)))
```

Comment:
indentation??

Reply:
Still indentation is off - please align `(IN_SET(...` with `j->unit->ignore_on_isolate` above.